addons/prometheus/0.59.1-40.1.0/install.sh

```diff
@@ -0,0 +1,144 @@
+
+    # remove services that had a clusterip change
+    kubectl delete service -n monitoring kube-state-metrics || true
+    kubectl delete service -n monitoring prometheus-operator || true
+
+    # remove nodeport services that had names change
+    kubectl delete service -n monitoring grafana || true
+    kubectl delete service -n monitoring alertmanager-main || true
+    kubectl delete service -n monitoring prometheus-k8s || true
+
+    # if the prometheus-node-exporter daemonset exists and has a release labelSelector set, delete it
+    if kubernetes_resource_exists monitoring daemonset prometheus-node-exporter; then
+        local promNodeExporterLabelSelector=$(kubectl get daemonset -n monitoring prometheus-node-exporter --output="jsonpath={.spec.selector.matchLabels.release}")
+        if [ -n "$promNodeExporterLabelSelector" ]; then
+            kubectl delete daemonset -n monitoring prometheus-node-exporter || true
+        fi
+    fi
+
+    # if the prometheus-operator deployment exists and has the wrong labelSelectors set, delete it
+    if kubernetes_resource_exists monitoring deployment prometheus-operator; then
+        local promOperatorLabelSelector=$(kubectl get deployment -n monitoring prometheus-operator --output="jsonpath={.spec.selector.matchLabels.release}") || true
+        if [ -n "$promOperatorLabelSelector" ]; then
+            kubectl delete deployment -n monitoring prometheus-operator || true
+        fi
+
+        promOperatorLabelSelector=$(kubectl get deployment -n monitoring prometheus-operator --output="jsonpath={.spec.selector.matchLabels.app\.kubernetes\.io/component}") || true
+        if [ -n "$promOperatorLabelSelector" ]; then
+            kubectl delete deployment -n monitoring prometheus-operator || true
+        fi
+    fi
+
+    # the metrics service has been renamed to v1beta1.custom.metrics.k8s.io, delete the old
+    if kubectl get --no-headers apiservice v1beta1.metrics.k8s.io 2>/dev/null | grep -q 'monitoring/prometheus-adapter' ; then
+        kubectl delete apiservice v1beta1.metrics.k8s.io
+    fi
+
+    # change ClusterIP services to NodePorts if required
+    if [ -z "$PROMETHEUS_SERVICE_TYPE" ] || [ "$PROMETHEUS_SERVICE_TYPE" = "NodePort" ] ; then
+        cp "$src/nodeport-services.yaml" "$operatordst"
+        insert_patches_strategic_merge "$operatordst/kustomization.yaml" nodeport-services.yaml
+    fi
+
+    kubectl apply -k "$operatordst/"
+}
+
+GRAFANA_ADMIN_USER=
+GRAFANA_ADMIN_PASS=
+function grafana_admin_secret() {
+    if kubernetes_resource_exists monitoring secret grafana-admin; then
+        return 0
+    fi
+
+    local src="$1"
+    local grafanadst="$2"
+
+    GRAFANA_ADMIN_USER=admin
+    GRAFANA_ADMIN_PASS=$(< /dev/urandom tr -dc A-Za-z0-9 | head -c9)
+
+    insert_resources "$grafanadst/kustomization.yaml" grafana-secret.yaml
+
+    render_yaml_file "$src/tmpl-grafana-secret.yaml" > "$grafanadst/grafana-secret.yaml"
+}
+
+function prometheus_outro() {
```

Comment:
Could not find where this is used.

Reply:
`ADDONNAME_outro` functions are called by the install loop

https://github.com/replicatedhq/kURL/blob/e1436961009c1a0d7bb31e7ec23c19641ed1573c/scripts/common/addon.sh#L187-L191

Reply:
Good!